### PR TITLE
feat: converting existing api to drf base api.

### PR DESCRIPTION
**Pull request by awais786**
platform proposed changes [document](https://docs.google.com/document/d/1s-Aqbq8ohA-YG9jhrV3VCQyAUXDQgE8nq4DH7Ntr79E/edit
).

**Switch all APIs to be using DRF with standard Auth classes.**

This api seems a good candidate to update this code [get_student_progress_url](https://github.com/openedx/edx-platform/blob/63e940d65dbcd0191aa3d2865de6420d0f2b2c33/lms/djangoapps/instructor/views/api.py#L1728)

### CODE REVIEW

File: lms/djangoapps/instructor/permissions.py [hunk 1]
```diff
@@ -4,8 +4,11 @@
 
 from bridgekeeper import perms
 from bridgekeeper.rules import is_staff
+from opaque_keys.edx.keys import CourseKey
+from rest_framework.permissions import BasePermission  # lint-amnesty, pylint: disable=wrong-import-order
```
Comment (ziafazal) [inline lms/djangoapps/instructor/permissions.py:8]: can we remove lint-amnesty and order the import in appropriate way?

File: lms/djangoapps/instructor/permissions.py [hunk 2]
```diff
@@ -72,3 +75,11 @@
...
 perms[VIEW_ENROLLMENTS] = HasAccessRule('staff')
 perms[VIEW_FORUM_MEMBERS] = HasAccessRule('staff')
+
+
+class InstructorPermission(BasePermission):
+    """Generic permissions"""
+    def has_permission(self, request, view):
+        course = get_course_by_id(CourseKey.from_string(view.kwargs.get('course_id')))
+        permission = getattr(view, 'custom_permission', None)
+        return request.user.has_perm(permission, course)
```
Comment (ziafazal) [inline lms/djangoapps/instructor/permissions.py:85]: Can we write few unit tests to support this permission class?

File: lms/djangoapps/instructor/views/api.py [hunk 3]
```diff
@@ -1717,6 +1721,61 @@ def get_student_enrollment_status(request, course_id):
...
+
+        return progress_url
+
+
+@method_decorator(cache_control(no_cache=True, no_store=True, must_revalidate=True), name='dispatch')
+class StudentProgressUrl(APIView):
+    """
+    Get the progress url of a student.
+    Limited to staff access.
+    Takes query parameter unique_student_identifier and if the student exists
```
Comment (ziafazal) [inline lms/djangoapps/instructor/views/api.py:1758]: We should have very detailed docstring for APIs having these segments 

**Endpoint Description**: Clearly state what the endpoint does and its purpose in the API.

**HTTP Method**: Specify the HTTP method (GET, POST, PUT, DELETE, etc.) supported by the endpoint.

**URL Structure**: Provide the URL structure of the endpoint, including any path parameters or query parameters.

**Request Parameters**: List all parameters that can be included in the request (both required and optional), including their data types and formats.

**Authentication**: Explain any authentication mechanism required to access the endpoint (e.g., API key, OAuth token).

**Response**: Describe the structure and format of the response returned by the endpoint, including HTTP status codes and any error messages.

**Example Requests and Responses**: Provide sample requests and responses to illustrate how the endpoint should be used.

**Rate Limiting**: If applicable, mention any rate limiting or throttling applied to the endpoint.

File: lms/djangoapps/instructor/views/api.py [hunk 4]
```diff
@@ -1717,6 +1721,61 @@ def get_student_enrollment_status(request, course_id):
...
+    returns e.g. {
+        'progress_url': '/../...'
+    }
+    """
+    permission_classes = [IsAuthenticated, permissions.InstructorPermission]
```
Comment (ziafazal) [inline lms/djangoapps/instructor/views/api.py:1763]: We should add appropriate authentication class

File: lms/djangoapps/instructor/views/api.py [hunk 5]
```diff
@@ -1717,6 +1721,61 @@ def get_student_enrollment_status(request, course_id):
     return JsonResponse(response_payload)
 
 
+class StudentProgressUrlSerializer(serializers.Serializer):
+    """Serializer for course renders"""
+    unique_student_identifier = serializers.CharField(write_only=True)
+    course_id = CourseKeyField(required=False)
```
Comment (ziafazal) [inline lms/djangoapps/instructor/views/api.py:1727]: why `required` is set to `False` whereas course_id is needed to get progress url?

**Review by ziafazal — CHANGES_REQUESTED** [review]
@awais786 Thanks for making these changes I have few suggestions it would be great if you could look into those.

File: lms/djangoapps/instructor/views/api.py (hunk 5 above)
Comment (awais786) [reply lms/djangoapps/instructor/views/api.py:1727]: Its coming from URL as `kwarg`

File: lms/djangoapps/instructor/views/api.py (hunk 4 above)
Comment (awais786) [reply lms/djangoapps/instructor/views/api.py:1763]: I simply matched the old behaviour.

**Pull request closed without merging**